cmd/boostx/utils_cmd.go

```diff
@@ -281,3 +334,61 @@ var generatecarCmd = &cli.Command{
 		return nil
 	},
 }
+
+func signAndPushToMpool(ctx context.Context, api api.Gateway, n *clinode.Node, msg *types.Message) (cid cid.Cid, sent bool, err error) {
+	ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
+	messagesigner := messagesigner.NewMessageSigner(n.Wallet, &modules.MpoolNonceAPI{ChainModule: api, StateModule: api}, ds)
+
+	head, err := api.ChainHead(ctx)
+	if err != nil {
+		return
+	}
+	basefee := head.Blocks()[0].ParentBaseFee
+
+	spec := &lapi.MessageSendSpec{
+		MaxFee: abi.NewTokenAmount(1000000000), // 1 nFIL
+	}
+	msg, err = api.GasEstimateMessageGas(ctx, msg, spec, types.EmptyTSK)
+	if err != nil {
+		err = xerrors.Errorf("GasEstimateMessageGas error: %w", err)
+		return
+	}
+
+	// use basefee + 20%
+	newGasFeeCap := big.Mul(big.Int(basefee), big.NewInt(6))
+	newGasFeeCap = big.Div(newGasFeeCap, big.NewInt(5))
+
+	if big.Cmp(msg.GasFeeCap, newGasFeeCap) < 0 {
+		msg.GasFeeCap = newGasFeeCap
+	}
+
+	smsg, err := messagesigner.SignMessage(ctx, msg, func(*types.SignedMessage) error { return nil })
+	if err != nil {
+		return
+	}
+
+	fmt.Println("about to send message with the following gas costs")
+	maxFee := big.Mul(smsg.Message.GasFeeCap, big.NewInt(smsg.Message.GasLimit))
+	fmt.Println("max fee:     ", types.FIL(maxFee), "(absolute maximum amount you are willing to pay to get your transaction confirmed)")
```

Comment:
Note that `gas fee cap` is per gas unit, whereas here I also added `max fee` for easier reasoning.